server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -205,8 +205,8 @@ private void startNewSnapshots(SnapshotsInProgress snapshotsInProgress) {
                     // Add all new shards to start processing on
                     final ShardId shardId = shard.key;
                     final ShardSnapshotStatus shardSnapshotStatus = shard.value;
-                    if (localNodeId.equals(shardSnapshotStatus.nodeId())
-                        && shardSnapshotStatus.state() == ShardState.INIT
+                    if (shardSnapshotStatus.state() == ShardState.INIT
```

Comment:
This is executed on all shard of all snapshots so it's super hot for large clusters, much cheaper to check state before doing the more expensive node id check.